### PR TITLE
Update workflows templates

### PR DESCRIPTION
**Pull request by EyalDelarea**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---

Update workflows templates for create-fix-pull-requests for GitHub & GitLab.
Removing the push trigger and setting secluded job on specific branches.

### CODE REVIEW

File: docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-maven.yml [hunk 1]
```diff
@@ -16,103 +10,109 @@ permissions:
...
-          # Handle vulnerabilities with fix versions only
-          # JF_FIXABLE_ONLY: "TRUE"
-
-          # [Optional]
-          # Set the minimum severity for vulnerabilities that should be fixed and commented on in pull requests
-          # The following values are accepted: Low, Medium, High or Critical
-          # JF_MIN_SEVERITY: ""
+    strategy:
+      matrix:
+        # Creating fix pull requests will be triggered periodically on the following branches.
```
Comment (eyalbe4) [inline docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-maven.yml:15]: Creating fix pull requests will be triggered periodically on the following branches.
-->
The repository scanning will be triggered periodically on the following branches.

(let's change in all templates).

File: docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-npm.yml [hunk 2]
```diff
@@ -1,11 +1,5 @@
 name: "Frogbot Scan and Fix"
 on:
-  push:
-    # Creating fix pull requests will be triggered by any push to one of these branches.
-    # You can add or replace to any branch you want to open fix pull requests for.
-    branches:
-      - "main"
-      - "master"
   schedule:
     # The job will run once a day at 00:00 GMT.
```
Comment (eyalbe4) [inline docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-npm.yml:4]: The job will run once a day at 00:00 GMT.
-->
The repository will be scanned once a day at 00:00 GMT.

(let's change in all templates).

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-pipenv.yml [hunk 3]
```diff
@@ -16,109 +10,115 @@ permissions:
...
+            ref: ${{ matrix.branch }}
+
+        # Install prerequisites
+        - name: Setup Python
+          uses: actions/setup-python@v4
+          with:
+            python-version: '3.9'
+            cache: 'pipenv'
+        - name: Install pipenv
+          run: curl https://raw.githubusercontent.com/pypa/pipenv/master/get-pipenv.py | python
```
Comment (sverdlov93) [inline docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-pipenv.yml:29]:
```suggestion
          run: python -m pip install pipenv
```

File: docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-poetry.yml [hunk 4]
```diff
@@ -16,104 +10,110 @@ permissions:
...
+        # The repository scanning will be triggered periodically on the following branches.
+        branch: [ "dev","main" ]
+      steps:
+        - uses: actions/checkout@v3
+          with:
+            ref: ${{ matrix.branch }}
+
+        # Install prerequisites
+        - name: Install poetry
+          run: pipx install poetry
```
Comment (sverdlov93) [inline docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-poetry.yml:24]:
```suggestion
          run: python -m pip install pipenv
```

File: docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-poetry.yml [hunk 5]
```diff
@@ -16,104 +10,110 @@ permissions:
...
+        - name: Setup Python
```
Comment (sverdlov93) [inline docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-poetry.yml:25]: should be before poetry installl

File: docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-npm.yml [hunk 6]
```diff
@@ -16,109 +10,115 @@ permissions:
...
-
-          # [Optional]
-          # Set the minimum severity for vulnerabilities that should be fixed and commented on in pull requests
-          # The following values are accepted: Low, Medium, High or Critical
-          # JF_MIN_SEVERITY: ""
\ No newline at end of file
+    strategy:
+      matrix:
+        # The repository scanning will be triggered periodically on the following branches.
+        branch: [ "dev","main" ]
```
Comment (sverdlov93) [inline docs/templates/github-actions/scan-and-fix/frogbot-scan-and-fix-npm.yml:16]:
```suggestion
        branch: [ "dev", "main" ]
```

File: starter-workflows/code-scanning/frogbot-scan-and-fix.yml [hunk 7]
```diff
@@ -23,104 +21,110 @@ permissions:
...
-          # JF_PULL_REQUEST_TITLE_TEMPLATE: "[🐸 Frogbot] Upgrade ${IMPACTED_PACKAGE} to to ${FIX_VERSION}"
-
-          # [Optional, Default: "FALSE"]
-          # If TRUE, Frogbot creates a single pull request with all the fixes.
-          # If FALSE, Frogbot creates a separate pull request for each fix.
-          # JF_GIT_AGGREGATE_FIXES: "FALSE"
+    strategy:
+      matrix:
+        # The repository scanning will be triggered periodically on the following branches.
+        branch: [ "dev","main" ]
```
Comment (sverdlov93) [inline starter-workflows/code-scanning/frogbot-scan-and-fix.yml:27]:
```suggestion
        branch: [ "dev", "main" ]
```

**Pull request merged**